Wordle.Api/Wordle.Api/Controllers/LeaderboardController.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Wordle.Api.Data;
+using Wordle.Api.Dtos;
+using Wordle.Api.Services;
+
+namespace Wordle.Api.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class LeaderboardController : ControllerBase
+    {
+        private readonly LeaderboardService _leaderboardService;
+
+        public LeaderboardController(LeaderboardService leaderboardService)
+        {
+            _leaderboardService = leaderboardService;
+        }
+
+        [HttpGet("GetTopPlayers")]
+        public async Task<IEnumerable<Player>> Get(int? count)
+        {
+            return await _leaderboardService.GetTopPlayers(count);
+        }
+
+        [HttpPost("InsertScore")]
+        public async Task<Player> AddScore(string name, int numAttempts, int seconds)
+        {
+            return await _leaderboardService.AddScore(name, numAttempts, seconds);
+        }
+
```

Comment:
Over here you are going to want to do exactly what we did in class with the Word (using FromBody) and all your problems will be solved. 
The reason you are having to concatenate a bunch of stuff together is because you set up AddScore to accept multiple parameters and therefore, on the front end, you cannot send a json object like you are wanting to. What you want to do is set up AddScore to take in an object so that you can send an object. That object will be a player but without the id and gamecount. 

So first create a DTO (data transfer object) for the Player and include all the properties that you would want to send the backend from the frontend. For example in my DTO I have the players name, number of attempts, and number of seconds to completion. Then change the AddScore method to take in "[FromBody] PlayerDto player" as a parameter and refactor. Should be good to go after that to send json objects to the backend.

Reply:
Thank you so much, I completely forgot about the DTO!  I'll be implementing the suggested changes tomorrow and Thursday.